### PR TITLE
Add automated periodic integration test coverage for SQLite

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
Setups up an external SQLite database based on the path in`test-credentials.yml`. Creates the parent directories too.

## Related issue number (if any)
ENG-2637

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: integration_tests/sdk/setup_integration.py [hunk 1]
```diff
@@ -142,10 +151,25 @@ def _setup_snowflake_data(client: Client, snowflake: RelationalDBIntegration) ->
     _add_missing_artifacts(client, snowflake, existing_table_names)
 
 
+def _setup_external_sqlite_db(path: str):
+    """Spins up an external SQLite database at 'path'."""
+    assert path[-1] != "/", "Path must point to a file"
```
Comment (likawind) [inline integration_tests/sdk/setup_integration.py:156]: I think `os.path` may have some API like `is_dir()`

File: integration_tests/sdk/setup_integration.py [hunk 2]
```diff
@@ -142,10 +151,25 @@ def _setup_snowflake_data(client: Client, snowflake: RelationalDBIntegration) ->
...
+
+    import os
+    from pathlib import Path
+
+    # Create the parent directories if they don't already exist.
+    db_abspath = os.path.expanduser(path)
+    db_dirpath = Path((os.path.dirname(db_abspath)))
+    db_dirpath.mkdir(parents=True, exist_ok=True)
```
Comment (likawind) [inline integration_tests/sdk/setup_integration.py:164]: I assume this handles dir creation recursively?

**Review by likawind — APPROVED** [review]
LGTM, some questions about the file path:
* I couldn't find the code that creates a default file if no sqlite DB file is provided, could you give a pointer to that?
* Do we need to worry about GC the created file, esp. in the above case?

**Review by kenxu95 — COMMENTED** [review]
I don't think we need to worry about garbage collecting, since this only runs on github actions and locally.

**Pull request merged**